docs: add summary_overview template (#1946)
googleapis/synthtool: synthtool/gcp/templates/python_library/docs/summary_overview.md

> [

Reviewer: Note, this file may need to be excluded in `owlbot.py` manually downstream for repositories that publish to googleapis.dev instead of c.g.c since they won't have the corresponding `summary_class.html`, `summary_method.html` and `summary_property.html`

Reviewer: After a closer look, we could exclude them in code by using `self.excludes += ["docs/summary_overview.md"]` when `is_google_cloud_api` is `False`.

Add an else block here for the split repo
https://github.com/googleapis/synthtool/blob/349943b54f15f9f9655a3d1c35d58acb68165678/synthtool/gcp/common.py#L299-L301

and here for the monorepo
https://github.com/googleapis/synthtool/blob/349943b54f15f9f9655a3d1c35d58acb68165678/synthtool/gcp/common.py#L235-L237

```
        # If the directory `google/cloud` exists, add kwargs to signal that the client library is for a Cloud API
        if Path("google/cloud").exists():
            kwargs["is_google_cloud_api"] = True
        else:
            self.excludes += ["docs/summary_overview.md"]
```

Author: Done.

This is a templated file. Adding content to this file may result in it being

> reverted. Instead, if you want to place additional content, use

Reviewer: ```suggestion
reverted. Instead, if you want to place additional content, create an
```

"overview_content.md" file in `docs/` directory. The Sphinx tool will

> pick up on the content and merge the content.

Reviewer: Once https://github.com/googleapis/sphinx-docfx-yaml/pull/362 is released, we can test this out. I think we should hold this PR until https://github.com/googleapis/sphinx-docfx-yaml/pull/362 is released.

Author: Done.

]: #

> # {{ metadata['repo']['name'] }} API
> 
> Overview of the APIs available for {{ metadata['repo']['name'] }} API.

Reviewer: ```suggestion
# {{ metadata['repo']['name_pretty'] }} API

Overview of the APIs available for {{ metadata['repo']['name_pretty'] }} API.
```

## All entries

> Take a look at all classes, methods and properties & attributes for

Reviewer: ```suggestion
Classes, methods and properties & attributes for
```

> {{ metadata['repo']['name'] }} API.

Reviewer: ```suggestion
{{ metadata['repo']['name_pretty'] }} API.
```

`name_pretty` is recommended as it reflects the product branding

[classes]({{ metadata['repo']['client_documentation'] }}/summary_class.html)

[methods]({{ metadata['repo']['client_documentation'] }}/summary_method.html)

[properties and

> attributes]({{ metadata['repo']['client_documentation'] }}/summary_property.html)

Reviewer: These files don't exist yet. Should we consider publishing them before we add an overview page?

Author: They were planned to be simultaneously published, but I've released new version of the plugin.
